### PR TITLE
Move pkg/kubectl/utils/{resource,qos} staging

### PR DESCRIPTION
**Pull request by sallyom**
* Moves pkg/kubectl/util/resource to staging.
* Publishes this code to the new kubectl repository: https://github.com/kubernetes/kubectl/pkg/util/resource
* Code will be available for re-use after publishing.
/kind cleanup
/sig cli
/area kubectl

```release-note
NONE
```

### CODE REVIEW

File: staging/publishing/rules.yaml [hunk 1]
```diff
@@ -629,9 +629,22 @@ rules:
       branch: master
       dir: staging/src/k8s.io/kubectl
     name: master
+    dependencies:
+    - repository: api
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: Can you also add kubectl to https://github.com/kubernetes/kubernetes/blob/master/staging/publishing/import-restrictions.yaml :)

File: staging/publishing/rules.yaml [hunk 2]
```diff
@@ -629,9 +629,22 @@ rules:
...
       dir: staging/src/k8s.io/kubectl
     name: release-1.15
     go: 1.12.5
-
+    dependencies:
+    - repository: api
+      branch: release-1.15
+    - repository: apimachinery
+      branch: release-1.15
+    - repository: client-go
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: From go.mod, it looks like client-go is not a dependency and can be removed?

Comment (sallyom) [reply staging/publishing/rules.yaml]: loooking, thanks.  there is a slew of these PRs atm (moving pkg/kubectl/util/* to staging), i _think_ we all are introducing this same change.. so whoever's merges first has to be correct lol

Comment (sallyom) [reply staging/publishing/rules.yaml]: looks like it, yes.  @damemi  @pswica @seans3

File: staging/publishing/rules.yaml (hunk 1 above)
Comment (sallyom) [reply staging/publishing/rules.yaml]: done, please check that I did that correctly, first time I've touched those files (lol this is my 2nd upstream PR) thanks :)

File: staging/publishing/rules.yaml (hunk 2 above)
Comment (sallyom) [reply staging/publishing/rules.yaml]: removed

**Comment by sallyom** [comment]
@nikhita I'm getting [this](https://github.com/kubernetes/kubernetes/pull/79202#issuecomment-504039093) with the 2 other PRs I've made the import-restrictions/publishing changes in, hm.. also, mind /ok-to-test this? thanks

File: staging/publishing/rules.yaml (hunk 1 above)
Comment (nikhita) [reply staging/publishing/rules.yaml]: > this is my 2nd upstream PR

Omg yay, welcome!! :tada: :tada:

Comment (sallyom) [reply staging/publishing/rules.yaml]: :blush

**Comment by sallyom** [comment]
@nikhita I think maybe i need to rebase then update-vendor, doing that now : )

File: staging/publishing/rules.yaml [hunk 3]
```diff
@@ -614,21 +614,22 @@ rules:
       dir: staging/src/k8s.io/kubectl
     name: master
     dependencies:
-    - repository: api
-      branch: master
-    - repository: apimachinery
-      branch: master
-    - repository: client-go
-      branch: master
+      - repository: api
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: The changes for this file need to be reverted, but looks good otherwise. :smile:

Comment (sallyom) [reply staging/publishing/rules.yaml]: thanks, I rebased too early, I've restored/fixed this.

**Comment by nikhita** [comment]
nit: can you squash the `Remove unused BUILD files` commits and `Updates to dependencies including go.mod, go.sum, and BUILD files` commits into a single commit? :grimacing:

**Comment by apelisse** [comment]
/approve

for `modules.txt`

**Comment by sallyom** [comment]
@nikhita i've squashed as you suggested.  thanks for your review : )

**Pull request merged**